src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt

```diff
@@ -1,34 +1,20 @@
 package org.imperial.mrc.hint.controllers
 
-import org.apache.tomcat.util.http.fileupload.FileUtils
-import org.imperial.mrc.hint.AppProperties
-import org.pac4j.core.config.Config
-import org.pac4j.core.context.WebContext
+import org.imperial.mrc.hint.FileManager
+import org.imperial.mrc.hint.FileType
 import org.springframework.web.bind.annotation.*
 import org.springframework.web.multipart.MultipartFile
-import java.io.File
 
 @RestController
 @RequestMapping("/baseline")
-class BaselineController(val context: WebContext,
-                         val pac4jConfig: Config,
-                         val appProperties: AppProperties) {
+class BaselineController(private val fileManager: FileManager) {
 
     @PostMapping("/pjnz/")
     @ResponseBody
     fun upload(@RequestParam("file") file: MultipartFile): String {
 
-        val id = pac4jConfig.sessionStore.getOrCreateSessionId(context)
         val fileName = file.originalFilename!!
-        val pjnzFile = File("${appProperties.uploadDirectory}/$id/pjnz/$fileName")
-
-        if (pjnzFile.parentFile.exists()) {
-            FileUtils.cleanDirectory(pjnzFile.parentFile)
-        } else {
-            FileUtils.forceMkdirParent(pjnzFile)
-        }
-
-        pjnzFile.writeBytes(file.bytes)
+        fileManager.saveFile(file, FileType.PJNZ)
```

Comment:
Does the pjnz not get validated too?

Reply:
Not yet because hintr isn't working, hence you can never progress to step 2. So for dev purposed I'm mocking this out still.